lib/parser/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.13)  # CMake version check
+project(simple_example)               # Create project "simple_example"
+set(CMAKE_CXX_STANDARD 17)            # Enable c++14 standard
+
+# Add main.cpp file of project root directory as source file
+set(SOURCE_FILES main.cpp Parser.hpp Parser.cpp Lexer.hpp Lexer.cpp ParserTests.cpp)
+
+# Add executable target with source files listed in SOURCE_FILES variable
+add_executable(ptx-parser ${SOURCE_FILES})
```

Comment:
Can you split this into 
1. A library - that contains the lexing + parsing capabilities
2. A tool (the ptx-parser) that lives in the tools directory at the top level>